docs/decisions/0012_clickhouse_dbt.rst

```diff
@@ -0,0 +1,65 @@
+12. Clickhouse vs dbt
+#####################
+
+Status
+******
+
+Accepted
+
+Context
+*******
+
+Most of the use cases for Aspects require complex transformations to be performed on the data stored in
+:ref:`clickhouse` before it can be displayed in :ref:`superset`. :ref:`xapi-concepts` is a good format for communicating
+learning-related events as they happen, but in order to analyse them and draw conclusions for what these events signify
+over time, Aspects needs views of the data that are designed specifically for the queries performed.
+
+:ref:`clickhouse` provides some features for transforming complex data, but creating efficient views and managing schema
+changes over time can be difficult. Plus, some of these features (like named collections) are not available on all
+hosted :ref:`clickhouse` services.
+
+This ADR describes the division of responsibilities in Aspects between the :ref:`clickhouse` database and :ref:`dbt`, a
+data transformation tool.
+
+Decision
+********
+
+**Store in Clickhouse, transform with dbt**
+
+Raw event and event sink data tables should continue to be created using :ref:`clickhouse-migrations`.
+
+Data transformations on this raw data should be made with :ref:`dbt`, including:
+
+* materialized views
+* partitions
+* dictionaries
+* fields extracted from event JSON
+
+Note that while we use dbt to manage these data transformations, the transformations themselves 
+
+Consequences
+************
+
+#. Contribute upstream to `dbt-clickhouse`_ where support for required features is missing.
+#. Move transformations made by the "query" and "dataset" Aspects Superset assets to `aspects-dbt`_.
```

Comment:
This probably deserves a "where possible", there are several things we still have to do in Superset virtual datasets because we have to use filters in specific ways for performance reasons, but I think this is still the long term goal as ClickHouse's predicate pushdown capabilities get overhauled.